Demo/src/app/main/main.component.ts

```diff
@@ -1,15 +1,44 @@
-import { Component, OnInit } from '@angular/core';
-
+import { Component } from '@angular/core';
+import {RegService} from "../Service/app.service";
+import {HttpErrorResponse} from "@angular/common/http";
+import {FormGroup, FormControlName, FormControl} from "@angular/forms";
+import{Router} from "@angular/router";
 @Component({
-  selector: 'app-main',
-  templateUrl: './main.component.html',
-  styleUrls: ['./main.component.css']
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
 })
-export class MainComponent implements OnInit {
-
-  constructor() { }
-
-  ngOnInit() {
+export class AppComponent {
+  private myForm:FormGroup;
+ private result:any;
+ constructor(private _service:RegService,
```

Comment:
do not use generic names, try to use meaningful variable names

Reply:
ok